### PR TITLE
Issue 55

### PR DESCRIPTION
**Pull request by laramiel**
This builds on my prior pull requests.

In addition, it update the google cloud logging plugin to use the google-api > 0.9, which resolves Issue #55.

### CODE REVIEW

**Comment by laramiel** [comment]
Updated the mime-types gemspec.

Note: This change removes the private_key auth method (see my remove_pk_auth branch), since google-api > 0.9 is not compatible with the old authentication method.

Perhaps making a release 0.4.16 or something would be a useful change before merging this?

**Comment by laramiel** [comment]
Well, that fails on google-api-client 0.9 and above due to the use of __dir__.

https://github.com/google/google-api-ruby-client/issues/387

**Comment by mr-salty** [comment]
Heh, we just can't win.  Well, maybe it's time to show 1.9 the door - I don't have a good sense for how many people still use it.

**Comment by mr-salty** [comment]
Also, on the "cutting a new release", that's certainly a good idea but I can release at other than HEAD so it doesn't have to block this CL.

I also meant to say, google-fluentd itself ships with ruby 2.1 (as does td-agent) so 1.9 is not a concern there...   it's only if someone does their own ruby install and installs fluentd "from scratch" that 1.9 even becomes an issue.  And, 1.9 has been officially EOL for over a year: https://www.ruby-lang.org/en/news/2014/01/10/ruby-1-9-3-will-end-on-2015/

**Comment by laramiel** [comment]
Then time to remove 1.9.x from Travis?

File: lib/fluent/plugin/out_google_cloud.rb [hunk 1]
```diff
@@ -117,6 +97,13 @@ class GoogleCloudOutput < BufferedOutput
     #   }
     config_param :label_map, :hash, :default => nil
 
+    # DEPRECATED: The following parameters, if present in the config
+    # indicate that the plugin configuration must be updated.
+    config_param :auth_method, :string, :default => nil
+    config_param :private_key_email, :string, :default => nil
+    config_param :private_key_path, :string, :default => nil
+    config_param :private_key_passphrase, :string, :default => nil
```
Comment (mr-salty) [inline lib/fluent/plugin/out_google_cloud.rb]: keep ```:secret => true``` here which prevents its value from being logged.

File: test/plugin/test_out_google_cloud.rb [hunk 2]
```diff
@@ -117,9 +118,9 @@ def setup
 
   # rubocop:disable Metrics/LineLength
   PRIVATE_KEY_CONFIG = %(
```
Comment (mr-salty) [inline test/plugin/test_out_google_cloud.rb]: on line 1215 there's a http stub with a comment "Used for 'private_key' auth." which I presume we can remove (sorry, github won't let me comment on the actual line).

**Comment by mr-salty** [comment]
I removed support for ruby 1.9 in  https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/pull/61 - PTAL and merge that.  LGTM otherwise - thanks!

File: fluent-plugin-google-cloud.gemspec [hunk 3]
```diff
@@ -14,7 +14,7 @@ eos
   gem.authors       = ['Todd Derr', 'Alex Robinson']
   gem.email         = ['salty@google.com']
 
-  gem.required_ruby_version = '~> 2.0'
```
Comment (laramiel) [inline fluent-plugin-google-cloud.gemspec]: I think that this works as well as the new one.

Comment (mr-salty) [inline fluent-plugin-google-cloud.gemspec]: oops - I didn't notice you had that in there.  I looked at a few other gems which used the Requirement::new syntax (despite using the shorter syntax later in the gemspec), but I wasn't able to determine the justification for doing it.

Either way is fine with me but I'd rather use >= 2.0 than ~>

Comment (laramiel) [inline fluent-plugin-google-cloud.gemspec]: Sure. I have no problem with that.

**Comment by mr-salty** [comment]
LGTM - thanks again!

**Pull request merged**